plugins/liip/repaircafe/components/Button.php

```diff
@@ -0,0 +1,135 @@
+<?php namespace Liip\RepairCafe\Components;
+
+use Cms\Classes\ComponentBase;
+
+class Button extends ComponentBase
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function componentDetails()
+    {
+        return [
+            'name'        => 'liip.repaircafe::lang.component.button.details.name',
+            'description' => 'liip.repaircafe::lang.component.button.details.description',
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function defineProperties()
+    {
+        return [
+            'label' => [
+                'title'             => 'liip.repaircafe::lang.component.button.label.title',
+                'description'       => 'liip.repaircafe::lang.component.button.label.description',
+                'required'          => true,
+                'type'              => 'string',
+            ],
+            'url' => [
+                'title'             => 'liip.repaircafe::lang.component.button.url.title',
+                'description'       => 'liip.repaircafe::lang.component.button.url.description',
+                'required'          => true,
+                'type'              => 'string',
+            ],
+            'target' => [
+                'title'             => 'liip.repaircafe::lang.component.button.target.title',
+                'description'       => 'liip.repaircafe::lang.component.button.target.description',
+                'default'           => '',
+                'type'              => 'dropdown',
+                'options'           => [
+                    '' => 'liip.repaircafe::lang.component.button.target.option.self',
+                    '_blank' => 'liip.repaircafe::lang.component.button.target.option.blank',
+                ],
+            ],
+            'style' => [
+                'title'             => 'liip.repaircafe::lang.component.button.style.title',
+                'description'       => 'liip.repaircafe::lang.component.button.style.description',
+                'default'           => 'btn-primary',
+                'required'          => true,
+                'type'              => 'dropdown',
+                'options'           => [
+                    'btn-primary' => 'liip.repaircafe::lang.component.button.style.option.btn_primary',
+                    'btn-secondary' => 'liip.repaircafe::lang.component.button.style.option.btn_secondary',
+                    'btn-outline-primary' => 'liip.repaircafe::lang.component.button.style.option.btn_outline_primary',
+                    'btn-outline-secondary' => 'liip.repaircafe::lang.component.button.style.option.btn_outline_secondary',
+                ],
+            ],
+            'position' => [
+                'title'             => 'liip.repaircafe::lang.component.button.position.title',
+                'description'       => 'liip.repaircafe::lang.component.button.position.description',
+                'default'           => '',
```

Comment:
Do you think it could make sense to set a default here? Just to be safe?

Reply:
It is.. the default value is the '' (empty) option ;) -> see options

Reply:
Yeah, I saw that, just wondered if that should actually be empty, but should be fine :)